cmd/gapit/trace2apk.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/app/layout"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/os/device"
+	"github.com/google/gapid/gapis/service"
+)
+
+type trace2apkVerb struct{ Trace2apkFlags }
+
+func init() {
+	verb := &trace2apkVerb{}
+	app.AddVerb(&app.Verb{
+		Name:      "trace2apk",
+		ShortHelp: "Create an stand-alone APK which replays a trace (off-screen)",
+		Action:    verb,
+	})
+}
+
+func (verb *trace2apkVerb) Run(ctx context.Context, flags flag.FlagSet) error {
+	if flags.NArg() != 1 {
+		app.Usage(ctx, "Exactly one gfx trace file expected, got %d", flags.NArg())
+		return nil
+	}
+
+	client, capturePath, err := getGapisAndLoadCapture(ctx, verb.Gapis, verb.Gapir, flags.Arg(0), verb.CaptureFileFlags)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	if err := client.ExportReplay(ctx, capturePath, nil, "assets/export_replay", &service.ExportReplayOptions{}); err != nil {
+		return log.Err(ctx, err, "Failed to export replay")
+	}
+
+	refAPKPath, err := layout.GapidApk(ctx, device.AndroidARM64v8a)
+	if err != nil {
+		return err
+	}
+
+	newAPK := strings.TrimSuffix(flags.Arg(0), ".gfxtrace") + ".replay.apk"
+	log.I(ctx, "New apk: %s", newAPK)
+
+	// Copy the apk to current dir with new name
+	err = fileCopy(newAPK, refAPKPath.String())
+	if err != nil {
+		return err
+	}
+
+	// TODO(hevrard): implement the addition of assets, manifest package renaming (still todo!), re-sign, re-zip in core/os/android/apk module
+
+	// aapt add assets
+	execCommand(ctx, "aapt", "add", "-v", newAPK, "assets/export_replay/resources.data", "assets/export_replay/resources.index", "assets/export_replay/payload.bin")
+	user, err := user.Current()
+	if err != nil {
+		return err
+	}
+
+	keystore := filepath.Join(user.HomeDir, ".android/debug.keystore")
```

Comment:
What should we do if this doesn't exist?
We may want to consider creating our own keystore if this is the case.